src/passes/FloatTrap.h

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright 2017 WebAssembly Community Group participants
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+//
+// Pass that supports potentially-trapping float operations.
+//
+
+#ifndef wasm_passes_FloatTrap_h
```

Comment:
why float trap as a name - doesn't this handle int div/rem of 0 also, etc.?

Reply:
That it does. How about `TrapModePass`?

Reply:
i'm terrible at names, but maybe `src/ast/trapping.h`, enum is `TrapMode`?